jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -634,15 +645,42 @@ def serialize_script(script):
     else:
         return result
 
-
 def mk_multisig_script(pubs, k):
     """ Given a list of pubkeys and an integer k,
     construct a multisig script for k of N, where N is
     the length of the list `pubs`; script is returned
     as hex string.
     """
-    return serialize_script([k] + pubs + [len(pubs)]) + 'ae'
+    return serialize_script([k] + pubs + [len(pubs)]) \
+        + 'ae' #OP_CHECKMULTISIG
 
+def mk_freeze_script(pub, locktime):
+    """
+    Given a pubkey and locktime, create a script which can only be spent
+    after the locktime has passed using OP_CHECKLOCKTIMEVERIFY
+    """
+    if not isinstance(locktime, int):
+        raise TypeError("locktime must be int")
+    if not isinstance(pub, bytes):
+        raise TypeError("pubkey must be in bytes")
+    usehex = False
+    if not is_valid_pubkey(pub, usehex, require_compressed=True):
+        raise ValueError("not a valid public key")
+    pub = binascii.hexlify(pub).decode()
+    return (serialize_script([locktime])
```

Comment:
The processing here looks unnecessarily awkward. Serialize script should handle the integers correctly (and binary) and you shouldn't need these one by one hex conversions (with the [2:] especially), but just pass a list to serialize_script. See this example I used back in the day:

https://github.com/AdamISZ/CoinSwapCS/blob/master/coinswap/base.py#L158-L163

I'll look into this myself (even if it's already clear/certain to you, it's good if someone else tries to draw their own conclusion). Unfortunate, this is all much cleaner after py-bitcointx.

Reply:
After some investigation and tests, I believe this produces identical output and is cleaner. Please check if you agree:

```
def mk_freeze_script(pub, locktime):
 if not isinstance(locktime, int):
        raise TypeError("locktime must be int")
    if not isinstance(pub, bytes):
        raise TypeError("pubkey must be in bytes")
    if not is_valid_pubkey(pub, False, require_compressed=True):
        raise ValueError("not a valid public key")
    scr = [locktime, btc.OP_CHECKLOCKTIMEVERIFY, btc.OP_DROP, pub, btc.OP_CHECKSIG]
    return binascii.hexlify(serialize_script(scr)).decode()
```

Reply:
Good code. I agree.

I actually tried for a bit to create code as clean as this but couldn't get it to work. Vitalik's mixing of bytes and string in the old pybitcointools is really crappy isn't it (not that we don't already know this).